### PR TITLE
add some missing or incorrect types to peewee

### PR DESCRIPTION
**Pull request by dhorkel**
I discovered some return types in for peewee were incorrect (reported here: https://github.com/python/typeshed/issues/12250). I decided to add some more types where it seemed straightforward.

This is my first attempt to contribute so apologies for any mistakes :)

### CODE REVIEW

File: stubs/peewee/peewee.pyi [hunk 1]
```diff
@@ -150,12 +150,12 @@ class _HashableSource:
...
     def __hash__(self) -> int: ...
-    def __eq__(self, other) -> bool: ...
-    def __ne__(self, other) -> bool: ...
-    __lt__: Incomplete
-    __le__: Incomplete
-    __gt__: Incomplete
-    __ge__: Incomplete
+    def __eq__(self, other) -> Expression | bool: ...
+    def __ne__(self, other) -> Expression | bool: ...
+    __lt__: Expression
```
Comment (JelleZijlstra) [inline stubs/peewee/peewee.pyi:155]: This seems incorrect, I assume `__lt__` should be a function returning an `Expression`, not an `Expression` itself.

Comment (dhorkel) [reply stubs/peewee/peewee.pyi:155]: From peewee's source code here: https://github.com/coleifer/peewee/blob/2b2e876edf686f3f4ddf5f36df9b978f42504082/peewee.py#L909

Most of the comparisons return something from the `_e()` function:
```
    def _e(op):
        def inner(self, rhs):
            return Expression(self, op, rhs)
        return inner
    __lt__ = _e(OP.LT)
    __le__ = _e(OP.LTE)
    __gt__ = _e(OP.GT)
    __ge__ = _e(OP.GTE)

```

Comment (dhorkel) [reply stubs/peewee/peewee.pyi]: Oops, thank you. Changed to `Callable[[Any], Expression]`

**Comment by JelleZijlstra** [comment]
You may need to add some type ignores to fix the CI errors:

```
stubs/peewee/peewee.pyi:153: error: Return type "Expression | bool" of "__eq__" incompatible with return type "bool" in supertype "object"  [override]
stubs/peewee/peewee.pyi:154: error: Return type "Expression | bool" of "__ne__" incompatible with return type "bool" in supertype "object"  [override]
stubs/peewee/peewee.pyi:256: error: Return type "Expression" of "__eq__" incompatible with return type "bool" in supertype "object"  [override]
stubs/peewee/peewee.pyi:257: error: Return type "Expression" of "__ne__" incompatible with return type "bool" in supertype "object"  [override]
stubs/peewee/peewee.pyi:1477: error: Return type "bool" of "__eq__" incompatible with return type "Expression" in supertype "ColumnBase"  [override]
stubs/peewee/peewee.pyi:1478: error: Return type "bool" of "__ne__" incompatible with return type "Expression" in supertype "ColumnBase"  [override]
```

**Pull request merged**